onnxruntime/core/providers/plaidml/plaidml_execution_provider.cc

```diff
@@ -122,32 +122,43 @@ common::Status PlaidMLExecutionProvider::Compile(
           Ort::CustomOpApi ort{*api};
           // TODO (PlaidML): why is this using custom op api? is it necessary ? is there a neater way?
           auto pml_state = static_cast<PlaidMLFunctionState*>(state);
-          auto binder = plaidml::exec::Binder(*pml_state->program->program);
-
+          pml_state->program->program->compile();
           // Load input data
-          auto executable = binder.compile();
+          auto executable = plaidml::exec::Executable(*pml_state->program->program);
+
           unsigned input_idx = 0;
+          std::vector<plaidml::Buffer> inputs;
+          std::vector<plaidml::Buffer> outputs;
+
+          for (auto input_arg : pml_state->program->program->inputs()) {
+            inputs.emplace_back(plaidml::Buffer(input_arg));
+          }
+          for (auto output_arg : pml_state->program->program->outputs()) {
+            outputs.emplace_back(plaidml::Buffer(output_arg));
+          }
+          //auto input_shapes = pml_state->program->program->inputs();
           for (auto input_placeholder : pml_state->program->get_program_inputs()) {
             // program->inputs and ORT inputs are in the same order, so these match
             // TODO (PlaidML): check if a lookup method is required here or it is sufficient to reply
             // on the matching input order
             const OrtValue* input_value = ort.KernelContext_GetInput(context, input_idx++);
             void* input_data = const_cast<void*>(ort.GetTensorData<void>(input_value));
-            binder.input(input_placeholder).copy_from(input_data);
+            inputs[input_idx-1].copy_from(input_data);
```

Comment:
IMO it's slightly cleaner to move the ++ to the 2nd statement and skip the -1

Reply:
Better yet, maybe just construct the inputs in this loop rather than creating an empty buffer in the previous loop. Basically construct the buffer with shape and data together.